### PR TITLE
Issue-6289 Copy node flipbook animation when cloning

### LINKED ISSUE #6289: No animation when you use Gui.clone.
**when you use Gui.clone it doesn’t copy the animation.**

I need use gui.play_flipbook(node, anim) for animate the clone node.

In this GIF, the original node is at right and the clone node at left. Within use gui.play_flipbook for the clone node.
tuffer

**More details :**
https://forum.defold.com/t/gui-clone-not-use-animation-group/69992

  - Defold version 1.2.190

### PR DESCRIPTION
**Pull request by britzl**
Fixes #6289

### CODE REVIEW

File: engine/gui/src/test/test_gui.cpp [hunk 1]
```diff
@@ -6290,6 +6290,42 @@ TEST_F(dmGuiTest, ScriptGetSetAlpha)
...
+    // same texture on the node?
+    ASSERT_EQ(dmGui::GetNodeTextureId(m_Scene, node), dmGui::GetNodeTextureId(m_Scene, clone));
+
+    // same playback rate?
+    ASSERT_EQ(dmGui::GetNodeFlipbookPlaybackRate(m_Scene, node), dmGui::GetNodeFlipbookPlaybackRate(m_Scene, clone));
+
+    // same animation?
+    ASSERT_EQ(dmGui::GetNodeFlipbookAnimId(m_Scene, node), dmGui::GetNodeFlipbookAnimId(m_Scene, clone));
+
+    // cleanup
```
Comment (britzl) [inline engine/gui/src/test/test_gui.cpp:6324]: Is this needed for the tests?

Comment (JCash) [reply engine/gui/src/test/test_gui.cpp:6324]: Not sure tbh.
Perhaps we have a leak detection in the engine, but I don't think so 🤔

**Review by JCash — APPROVED** [review]

File: engine/gui/src/gui.cpp [hunk 2]
```diff
@@ -4443,6 +4443,12 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
             SetNodeSpineScene(scene, *out_node, GetNodeSpineScene(scene, node), GetNodeSpineSkin(scene, node), GetNodeSpineAnimation(scene, node), false);
         }
 
+        if (n->m_Node.m_FlipbookAnimHash != 0)
+        {
+            float playback_rate = GetNodeFlipbookPlaybackRate(scene, node);
+            PlayNodeFlipbookAnim(scene, *out_node, n->m_Node.m_FlipbookAnimHash, 0.0f, playback_rate, 0, 0, 0);
```
Comment (JCash) [inline engine/gui/src/gui.cpp:4449]: I would assume that a clone should copy the current time + playback mode as well?

**Review by JCash — CHANGES_REQUESTED** [review]
I think some more state needs to be copied?

File: engine/gui/src/gui.cpp (hunk 2 above)
Comment (britzl) [reply engine/gui/src/gui.cpp:4449]: Perhaps current time, but playback mode can't be changed at runtime?

**Review by JCash — APPROVED** [review]

**Pull request merged**